### PR TITLE
improve string entry handling

### PR DESCRIPTION
**Pull request by florian-huber**
- refactored `add_precursor_mz()`
- refactored `ModifiedCosine`
- added logging messages and allow `ModifiedCosine` to also handle string entries for precursor_mz.

### CODE REVIEW

**Comment by florian-huber** [comment]
This is to address #273. Started small but became a little bit bigger. @hechth let me know if want/can have a brief look at it. If not, just let me know and I will do a self-review on this one.

**Comment by hechth** [comment]
If until next week is fine, I'll be happy to review it, I should find time tomorrow.

**Comment by florian-huber** [comment]
Sure, now hurry. And thanks a lot!

**Comment by florian-huber** [comment]
ping @hechth 
(feel free to skip if it's too busy!)

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,9 +9,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Added
 
+- better, more flexible string handling of `ModifiedCosine` [275](https://github.com/matchms/matchms/pull/275)
```
Comment (hechth) [inline CHANGELOG.md]:
```suggestion
- better, more flexible string handling of `ModifiedCosine` [#275](https://github.com/matchms/matchms/pull/275)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,9 +9,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - matchms logger, replacing all former `print` statments to better control logging output [#271](https://github.com/matchms/matchms/pull/271)
 - `add_logging_to_file()`, `set_matchms_logger_level()`, `reset_matchms_logger()` functions to adapt logging output to user needs [#271](https://github.com/matchms/matchms/pull/271) 
 
+## Changed
+
+- refactored `add_precursor_mz`, including better logging [275](https://github.com/matchms/matchms/pull/275)
```
Comment (hechth) [inline CHANGELOG.md]:
```suggestion
- refactored `add_precursor_mz`, including better logging [#275](https://github.com/matchms/matchms/pull/275)
```

File: q [hunk 3]
```diff
@@ -0,0 +1,427 @@
+CHANGELOG.md[36m:[m- `add_[1;31mprecursor[m_mz()` filter now also checks for metadata in keys `[1;31mprecursor[mmz` and `[1;31mprecursor[m_mass` [#223](https://github.com/matchms/matchms/pull/223)
```
Comment (hechth) [inline q]: This file I think should not be here!

**Review by hechth — COMMENTED** [review]
One log file or some other binary seemed to have gotten into the PR - maybe delete the file?

File: q (hunk 3 above)
Comment (florian-huber) [reply q]: Thanks! Good catch, I removed it.

**Comment by florian-huber** [comment]
Thanks a lot for reviewing Helge!

**Pull request merged**